src/goto-symex/symex_goto.cpp

```diff
@@ -108,7 +108,10 @@ void goto_symext::symex_goto(statet &state)
   }
 
   exprt simpl_state_guard = state.guard.as_expr();
+
+#ifndef BDD_GUARDS
   do_simplify(simpl_state_guard);
+#endif
```

Comment:
I'd recommend changing the `.as_expr()` interface to take a Boolean argument whether or not to run the simplifier (inside `.as_expr`). It could have a default value (`false`) so as not to break existing APIs.

Reply:
I don't like adding arguments that would always be called with a constant. An alternative would be to define a const static field of the guardt class that tells us whether guards benefit from simplifications.  That way an `if` condition on this Boolean should even get simplified by the compiler.

Reply:
Sounds like a good approach to me!